src/pages/workspaces/workspace/_tests/WorkspaceMenu.test.js

```diff
@@ -0,0 +1,258 @@
+import '@testing-library/jest-dom'
+
+import { fireEvent, render, screen } from '@testing-library/react'
+import { useState } from 'react'
+import { div, h } from 'react-hyperscript-helpers'
+import { MenuTrigger } from 'src/components/PopupTrigger'
+import TooltipTrigger from 'src/components/TooltipTrigger'
+import { useWorkspaceDetails } from 'src/components/workspace-utils'
+import WorkspaceMenu from 'src/pages/workspaces/workspace/WorkspaceMenu'
+
+
+jest.mock('src/components/workspace-utils', () => {
+  const originalModule = jest.requireActual('src/components/workspace-utils')
+  return {
+    ...originalModule,
+    useWorkspaceDetails: jest.fn()
+  }
+})
+
+jest.mock('src/components/PopupTrigger', () => {
+  const originalModule = jest.requireActual('src/components/PopupTrigger')
+  return {
+    ...originalModule,
+    MenuTrigger: jest.fn()
+  }
+})
+
+jest.mock('src/components/TooltipTrigger', () => ({
+  ...jest.requireActual('src/components/TooltipTrigger'),
+  __esModule: true,
+  default: jest.fn()
+}))
+
+
+describe('WorkspaceMenu - undefined workspace', () => {
+  beforeEach(() => {
+    useWorkspaceDetails.mockReturnValue({ workspace: undefined })
+    MenuTrigger.mockImplementation(({ content }) => { return div([content]) })
+    TooltipTrigger.mockImplementation(({ content, children }) => {
+      const [open, setOpen] = useState(false)
+      return (div([
+        div(
+          {
+            onMouseEnter: () => {
+              setOpen(true)
+            },
+            onMouseLeave: () => {
+              setOpen(false)
+            }
+          },
+          [children]
+        ),
+        open && !!content && div([content])
+      ]))
+    })
+  })
+
+  it('renders Clone menu item as disabled', () => {
+    const onClone = jest.fn()
+    render(h(WorkspaceMenu, {
+      iconSize: 20, popupLocation: 'left',
```

Comment:
Do you actually need to specify `iconSize`, `popupLocation`, and `callbacks`? We aren't actually triggering the callback function during the tests, as we are only checking the enable and tooltip states.

Reply:
Because `iconSize` and `popupLocation` do not have default values and are referenced, they do need to be specified. If possible, leave callbacks as empty though so it doesn't make it appear that the callback is actually tested.